packages/dds/task-manager/src/taskManager.ts

```diff
@@ -658,12 +660,23 @@ export class TaskManagerClass
 		this.connectionWatcher.emit("connect");
 	}
 
-	//
 	/**
 	 * Override resubmit core to avoid resubmission on reconnect.  On disconnect we accept our removal from the
 	 * queues, and leave it up to the user to decide whether they want to attempt to re-enter a queue on reconnect.
+	 * However, we do need to update latestPendingOps to account for the ops we will no longer be processing.
 	 */
-	protected reSubmitCore(): void {}
+	protected reSubmitCore(content: unknown, localOpMetadata: number): void {
+		assertIsTaskManagerOperation(content);
+		const pendingOps = this.latestPendingOps.get(content.taskId);
+		assert(pendingOps !== undefined, "No pending ops for task on resubmit attempt");
+		const pendingOp = pendingOps.shift();
```

Comment:
I always get a little fuzzy on the possible states where we could enter resubmit, but I think it's possible that we're asked to resubmit starting from an op that is not the first pending op (at least for staging mode commit?).  I like this strategy of updating pending ops during `reSubmitCore` though.

@anthony-murphy can probably say concretely, but you might need to find and splice rather than shift.

Reply:
Good callout. I think if it's not required it may be nice to keep it as a shift, but it's an easy change so I don't mind doing it preemptively either. Do you think we would have a similar concern with rollback?